### DISCUSSION TITLE
How to programmatically create and manage nodes?

### DISCUSSION THREAD

**Discussion by wbsmolen** [nikitabobko/AeroSpace#624] [questions-and-answers] [answered]
I'm trying to use my config file to drive actions that Aerospace should run on-startup. I have 3 monitors and I want to move a workspace to each monitor. Then I want Workspace 1 (on an ultrawide) to have 2 nodes, 1 horizontally tiled and 1 vertically tiled. 

Workspaces 2 and 3 have 1 node.

I want the config to use `exec-and-forget` to launch specific apps and place them in the workspaces. **This is where I'm having my problem.**

I'm trying to use `after-startup-command`: 

```
after-startup-command = [
    "move-workspace-to-monitor --workspace 3 --wrap-around next",
    "move-workspace-to-monitor --workspace 2 --wrap-around prev",
    

    "workspace 1",
    "layout horizontal",
    "exec-and-forget open -a '<browser app>'",  
    "exec-and-forget open -a '<browser app>' --args --new-window", 
    "exec-and-forget open -a '<browser app>' --args --new-window",  
    "exec-and-forget open -a '<some app>'", 
    "split vertical",  
    "exec-and-forget open -a '<some app>'",  
    "exec-and-forget open -a '<some app>'",  


    "workspace 2",
    "layout vertical",
    "exec-and-forget open -a '<some app>'",
    "exec-and-forget open -a '<some app>'",


    "workspace 3",
    "layout horizontal",
    "exec-and-forget open -a '<some app>'"
]
```

But this isn't working. I looked at using `exec-on-workspace-change` but that would require Aerospace exposing an api for node management so a foreign bash script could leverage it. I don't believe that exists.

## Checklist

- [x] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - [query 1](https://github.com/nikitabobko/AeroSpace/issues?q=programatically+manage+nodes)
  - [query 2](https://github.com/nikitabobko/AeroSpace/issues?q=manage+nodes)
  - [query 3](https://github.com/nikitabobko/AeroSpace/issues?q=node+api)
  - [query 4](https://github.com/nikitabobko/AeroSpace/issues?q=create+nodes+in+config)

- [x] I've read the guide https://nikitabobko.github.io/AeroSpace/guide
- [x] I've searched in the documentation (https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands). My search queries were:
  - programmatically manage nodes
  - assign window to node
  - node api
```
$ aerospace --version
aerospace CLI client version: 0.15.2-Beta b6cf82771f245ab7349a93baf8709e171537ff58
AeroSpace.app server version: 0.15.2-Beta b6cf82771f245ab7349a93baf8709e171537ff58
```

**Comment by nikitabobko** [comment, marked as answer]
Opening windows/applications happens with the delay.

The use case that you describe is not about programmatically creating nodes, but about waiting till the windows appear and become accessible for manipulation

Duplicate of #226
